pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -76,107 +73,74 @@ private SelectionOperatorUtils() {
   private static final String DOUBLE_PATTERN = "###################0.0#########";
   private static final DecimalFormatSymbols DECIMAL_FORMAT_SYMBOLS = DecimalFormatSymbols.getInstance(Locale.US);
 
-  /**
-   * Expand <code>'SELECT *'</code> to select all columns with {@link IndexSegment}, order all columns alphabetically.
-   * (Inner segment)
-   *
-   * @param selectionColumns unexpanded selection columns (may contain '*').
-   * @param indexSegment index segment.
-   * @return expanded selection columns.
-   */
-  @Nonnull
-  public static List<String> getSelectionColumns(@Nonnull List<String> selectionColumns,
-      @Nonnull IndexSegment indexSegment) {
-    if (selectionColumns.size() == 1 && selectionColumns.get(0).equals("*")) {
-      List<String> allColumns = new LinkedList<>(indexSegment.getPhysicalColumnNames());
-      Collections.sort(allColumns);
-      return allColumns;
-    } else {
-      return selectionColumns;
-    }
-  }
-
-  /**
-   * Extract all related columns for a selection query with {@link IndexSegment}. (Inner segment)
-   *
-   * @param selection selection query.
-   * @param indexSegment index segment.
-   * @return all related columns.
-   */
-  @Nonnull
-  public static Set<String> extractSelectionRelatedColumns(@Nonnull Selection selection,
-      @Nonnull IndexSegment indexSegment) {
-    Set<String> selectionColumns = new HashSet<>(getSelectionColumns(selection.getSelectionColumns(), indexSegment));
-    if (selection.getSelectionSortSequence() != null) {
-      for (SelectionSort selectionSort : selection.getSelectionSortSequence()) {
-        selectionColumns.add(selectionSort.getColumn());
-      }
-    }
-    return selectionColumns;
-  }
+  private static final ThreadLocal<DecimalFormat> THREAD_LOCAL_INT_FORMAT =
```

Comment:
Same here, capture this change in the PR description (as it seems not related to the intention of the PR).